src/openads-appnexus/repository/PullingAdRepository.js

```diff
@@ -10,14 +11,22 @@ export default class PullingAdRepository {
   }
 
   find ({id}) {
-    return Promise.race([
-      this._waitForData({id}),
-      this._timeoutPromise()
-    ])
+    return Promise.resolve(id)
+      .then(id => this.has({id}) ? this.getValue({id}) : this._pulledData({id}))
+  }
+
+  _pulledData ({id}) {
```

Comment:
why does this method is named using past tense (pulled) instead of the present (pull)? IMHO it will be clearer if we call it __pullData_ as it is an action to be performed immediately afterwards.

Reply:
Changed for 'pullForData' as suggested, tx!